js/saveState.js

```diff
@@ -0,0 +1,42 @@
+import _ from "lodash";
+
+import { COOKIE_LAST_STATE, COOKIE_OMIT_FROM_STATE } from "./constants/AppConstants";
+
+// store the cookie here in case we're testing
+let lastCookie = "";
+
+function cookiesToMap (cookieStr) {
+  return _.chain(cookieStr.split(/;\s*/))
+          .map((keyValStr) => keyValStr.split(/=/))
+          .fromPairs()
+          .value();
+}
+
+export function restoreState() {
+	let cookie = lastCookie;
+	if (typeof document !== 'undefined') {
```

Comment:
Could simplify this to just ```if (document)```, and if you want to add extra checking to make sure the cookie exists, you could check for that as well ```if (document && document.cookie)```